lisp/poly-ein.el

```diff
@@ -71,9 +99,18 @@
    :around (symbol-function 'syntax-ppss)
    (apply-partially #'poly-ein--narrow-to-inner #'1-))
 
+  (add-function
+   :around (symbol-function 'pm--mode-setup)
+   (lambda (f &rest args)
```

Comment:
Why is this hack necessary? 

There is already an advice around `font-lock-fontify-buffer` and other fontification functions which inhibits font-lock during the entire polymode initialization.

Reply:
I generally [skirt polylock altogether](https://github.com/dickmao/emacs-ipython-notebook/blob/dd531b5664784debb7885a66ffb81a77de5079ef/lisp/poly-ein.el#L47-L50) in favor of using native jit-lock.  In doing so, I must commit some hacks.

Reply:
All that stuff is insanely intrusive and affects other modes. Changing global polymode hacks and relying on internals is not a sustainable way forward. If you think poly-lock has issues and things could be done better you can open an issue on polymode and we will see what could be done. At this moment I would be surprised if there are real benefits to what you do. Poly-lock does essentially what jit-lock does but in a simpler and less obfuscated by backward compatibillty way.

Can you open an issue on ein or polymode and explain what is needed and what is not working correctly? If there is a common need I will be more than happy to accommodate it on polymode side. I am surprised that someone needs 300+ lines of code and loads of advises to write a single polymode.

Reply:
> Poly-lock does essentially what jit-lock does but in a simpler and less obfuscated by backward compatibility way

I don't think opening an issue critiquing the clone-tweak approach towards poly-lock counterfeiting jit-lock will make much of a difference.  I am more at home incorporating native jit-lock, capturing whatever improvements get made to it, even if that means some terrible, terrible advises.

Before activating polymode for EIN users, I will make an effort to compartmentalize the intrusiveness.

Reply:
> I don't think opening an issue critiquing the clone-tweak approach towards poly-lock counterfeiting jit-lock will make much of a difference. 

What you just said is basically "polymode devs are idiots and cannot recognize an obvious benefit of reusing existing code". 

Piling a huge hack around the two engines in an unrelated package is not a good citizenship. You hacks are comparable in size to the entire poly-lock code base. How is this an improvement? If you are really interested in polymode progress then you should at least inform the maintainer about the issues that provoked this entire initiative of yours. 

If you really think you can achieve what poly-lock does with less code and entirely relying on the jit-lock then I would be more than interested to look at it, but read further. 

> I am more at home incorporating native jit-lock, capturing whatever improvements get made to it, 

I think you are missing the point. Jit-lock is a small (but obfuscated for historical reasons) support mode for font-lock. Most of if is not used because `jit-lock-defer-time` is nil by default. So it's basically 60lines of code in `jit-lock-fontify-now` and a bunch of simple extension functions. What poly-lock does is having its own `poly-lock-fontify-now` function and it's own extension. It does so completely in line with font-lock requirements. Poly-lock is a support mode, but your advises constitute a hack which would eventually rewrite the entire jit-lock-fontify-now anyhow. I believe (maybe wrongly) that the level of protection that poly-mode needs cannot be achieved by simple advises of jit-lock-fontify-now.

FWIW getting rid of the poly-lock is one of my long term goals anyhow, but polymode is not ready for that yet because it still doesn't work well with c-mode languages. The complexity of c-mode might require tweaks hard to envisage at this moment. Tweaks which would be much easier to develop as part of poly-lock rather than jit-lock.

> Before activating polymode for EIN users, I will make an effort to compartmentalize the intrusiveness.

Thanks, but I would very much prefer if you could stop hacking on polymode internals outside of polymode repo entirely. I would be more than happy to consider improvements to the polymode with a prompt and careful consideration.